moto/cognitoidp/models.py

```diff
@@ -1441,12 +1465,16 @@ def admin_user_global_sign_out(self, user_pool_id, username):
         user_pool = self.describe_user_pool(user_pool_id)
         self.admin_get_user(user_pool_id, username)
 
-        for token, token_tuple in list(user_pool.refresh_tokens.items()):
-            if token_tuple is None:
-                continue
-            _, username = token_tuple
-            if username == username:
```

Comment:
This always evaluates to true, as we were overriding the value of the user-supplied `username`